addon/components/frost-list.js

```diff
@@ -0,0 +1,127 @@
+import Ember from 'ember'
+const {Component} = Ember
+import computed from 'ember-computed-decorators'
+import layout from '../templates/frost-list'
+import {PropTypes} from 'ember-prop-types'
+import SlotsMixin from 'ember-block-slots'
+import _ from 'lodash'
+
+const FrostList = Component.extend(SlotsMixin, {
+
+  // == Dependencies ==========================================================
+
+  // == Properties ============================================================
+  classNames: ['frost-list'],
+  layout: layout,
+  records: Ember.computed.alias('model'), // TODO Log deprecation of model as an attribute
+
+  propTypes: {
+    alwaysUseDefaultHeight: PropTypes.bool,
+    defaultHeight: PropTypes.number,
+    hook: PropTypes.string,
+    scrollPosition: PropTypes.number,
+    showDetail: PropTypes.bool,
+    size: PropTypes.string
+  },
+
+  // == Computed Properties =====================================================
+
+  // Normalize the data - TODO when the API improves this can be removed
+  @computed('records.[]')
+  _records (records) {
+    if (Ember.isEmpty(records)) {
+      return []
+    }
+    return records.map(function (record) {
+      if (_.isString(record.get('record-type'))) {
+        return record
+      }
+      record.set('record-type', record.get('dimension'))
+      return record
+    })
+  },
+
+  @computed('_records.[]', 'selections.[]')
+  mappedRecords (_records, selections) {
+    return _records.map((record) => {
+      if (_.includes(selections, record)) {
```

Comment:
There is probably a much more efficient way of doing this - maybe based on index or id so that we only have to query one entry instead of a loop